### PR TITLE
UML-3671 rework find actor lpa

### PR DESCRIPTION
**Pull request by nickdavis2001**
# Purpose

_Rework findActorInLpa to use combined format_

## Approach

_Make findActorInLpa be able to take GetAttorneyStatus interface as an arg, while for now continuing to support existing interim SiriusLpa. Move GetAttorneyStatus onto Person to enable that to happen. Extensive refactor of unit tests including adding fixture methods, to make them manageable. Replicate unit tests for combined format alongside interim format_

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [ ] New event_codes have been documented on the [wiki page](https://opgtransform.atlassian.net/wiki/spaces/LSML2/pages/3277881441/Understanding+the+event+logs)
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] I have notified the Interaction Designer of any content changes so that appropriate screenshots/flow diagram changes can be made
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-api/app/src/App/src/Entity/Person.php [hunk 1]
```diff
@@ -35,4 +36,34 @@ public function getUid(): string
...
+    }
+
+    #[DoNotSerialize]
+    public function getSystemStatus(): bool|string
+    {
+        return $this->systemStatus;
+    }
+
+    #[DoNotSerialize]
+    public function getPostcode(): string
```
Comment (cooperaj) [inline service-api/app/src/App/src/Entity/Person.php:59]: What interface requires postcode and dob?

File: service-api/app/src/App/src/Service/Lpa/FindActorInLpa.php [hunk 2]
```diff
@@ -21,12 +24,12 @@ public function __construct(private GetAttorneyStatus $getAttorneyStatus, privat
     {
     }
 
-    public function __invoke(SiriusLpa $lpa, array $matchData): ?array
+    public function __invoke(OldSiriusLpa|Lpa $lpa, array $matchData): ?array
```
Comment (cooperaj) [inline service-api/app/src/App/src/Service/Lpa/FindActorInLpa.php:27]: Would prefer to see the interface being passed in here, no the Lpa. So something like `OldSiriusLpa|FindActorInLpaInterface`

File: service-api/app/src/App/src/Service/Lpa/FindActorInLpa.php [hunk 3]
```diff
@@ -67,7 +70,7 @@ private function checkForAttorneyMatch(GetAttorneyStatusInterface $attorney, arr
         return [null, null];
     }
 
-    private function checkDonorDetails(SiriusPerson $donor, array $matchData): array
+    private function checkDonorDetails(SiriusPerson|Person $donor, array $matchData): array
```
Comment (cooperaj) [inline service-api/app/src/App/src/Service/Lpa/FindActorInLpa.php:73]: Same here but probably a different interface?

File: service-api/app/src/App/src/Service/Lpa/SiriusPerson.php [hunk 4]
```diff
@@ -46,6 +46,16 @@ public function getCompanyName(): string
         return (string)$this->person['companyName'];
     }
 
+    public function getPostcode(): string
```
Comment (cooperaj) [inline service-api/app/src/App/src/Service/Lpa/SiriusPerson.php:49]: These should be the same interface required by the new code.

**Review by allenannom — APPROVED** [review]
amazing!

File: service-api/app/src/App/src/Service/Lpa/AccessForAll/AddAccessForAllLpa.php [hunk 5]
```diff
@@ -240,21 +211,21 @@ public function validateRequest(string $userId, array $matchData): array
             );
             throw new BadRequestException(
                 'LPA details do not match',
-                ['lpaRegDate' => $lpaData['registrationDate']]
+                ['lpaRegDate' => $lpa['registrationDate']]
```
Comment (cooperaj) [inline service-api/app/src/App/src/Service/Lpa/AccessForAll/AddAccessForAllLpa.php:214]: This probably will fail.

**Review by allenannom — APPROVED** [review]

**Review by allenannom — APPROVED** [review]

**Pull request merged**